### DISCUSSION TITLE
In a Blazor Hybrid app, where exactly is the IJSRuntime dependency registered in the app startup?

### DISCUSSION THREAD

**Discussion by egvijayanand** [dotnet/maui#20192] [Q&A]
In a Blazor Hybrid app, where exactly is the IJSRuntime dependency registered in the app startup? I'm trying for the WinUI app. Framework trying to resolve the instance of it, but there's nothing registered.

In the documentation [here](https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/dependency-injection?view=aspnetcore-8.0#default-services), it's mentioned as a framework registered service. If that's the case, how it's handled in the Blazor Hybrid app?

@Eilon Your inputs are needed.

**Comment by crui3er** [comment]
In a Blazor Hybrid app WebViewJSRuntime is registered with AddBlazorWebView extension method https://github.com/dotnet/aspnetcore/blob/main/src/Components/WebView/WebView/src/ComponentsWebViewServiceCollectionExtensions.cs#L26 which is called from AddMauiBlazorWebView extension method https://github.com/dotnet/maui/blob/main/src/BlazorWebView/src/SharedSource/BlazorWebViewServiceCollectionExtensions.cs#L37

**Reply by Eilon** [reply]
Not sure how I missed this, but yes that's right.

Probably the reason it's not being resolved is that it's a Scoped Service, meaning it is only available to code running within that scope. If you go to the top-level DI container then the service is not available.

Fortunately, we added an API that lets you run code within that scope, so give this a try: https://learn.microsoft.com/dotnet/api/microsoft.aspnetcore.components.webview.maui.blazorwebview.trydispatchasync?view=net-maui-8.0

In the `workItem` callback you get the `IServiceProvider` and can manually resolve whatever scoped services you want (such as `IJSRuntime`).
